### PR TITLE
Load from filesystem

### PR DESCRIPTION
**Pull request by simonszu**
Ì have added the ability to specify a sanctorale file from file system. The `query` command first assumes the given option string to be a filesystem path, and checks if a file exists at this location. If yes, it tries to load this file as a data file and dies if it isn't successful. 
If there does not exist a file at the given location, `query` tries to interpret the given string as the name of a predefined sanctorale file, and proceeds in the known behavior.

### CODE REVIEW

File: lib/calendarium-romanum/cli.rb [hunk 1]
```diff
@@ -11,11 +11,23 @@ class CLI < Thor
     def query(date_str=nil)
       I18n.locale = options[:locale]
 
-      data_file = Data[options[:calendar]]
-      if data_file.nil?
-        die! 'Invalid calendar. See subcommand `calendars` for valid options.'
+      # This block is for specifying a full filesystem path
```
Comment (igneus) [inline lib/calendarium-romanum/cli.rb]: Please remove all comments. The code is short and very straightforward, comments are not necessary.

File: lib/calendarium-romanum/cli.rb [hunk 2]
```diff
@@ -11,11 +11,23 @@ class CLI < Thor
...
+      if File.exist?(options[:calendar])
```
Comment (igneus) [inline lib/calendarium-romanum/cli.rb]: `options[:calendar]` - when it's repeated three times, it's worth introducing a variable.

File: lib/calendarium-romanum/cli.rb [hunk 3]
```diff
@@ -11,11 +11,23 @@ class CLI < Thor
...
+          sanctorale = SanctoraleLoader.new.load_from_file(options[:calendar])
+        rescue CalendariumRomanum::InvalidDataError
+          die! 'Invalid file format.'
+        end
+      else
+        # This block is for predefined calendars
+        data_file = Data[options[:calendar]]
+
+        if data_file.nil?
+          die! 'Invalid calendar. See subcommand `calendars` for valid options.'
```
Comment (igneus) [inline lib/calendarium-romanum/cli.rb]: If the user makes an error while entering calendar path
`calendariumrom query --calendar /path/with/a/typo`
the program will respond `Invalid calendar. See subcommand `calendars` for valid options.` The error message should take into account that it is not really clear if the user was trying to load a bundled calendar or custom calendar from a filesystem path.

File: lib/calendarium-romanum/cli.rb (hunk 1 above)
Comment (simonszu) [reply lib/calendarium-romanum/cli.rb]: Done

File: lib/calendarium-romanum/cli.rb (hunk 2 above)
Comment (simonszu) [reply lib/calendarium-romanum/cli.rb]: Done

**Comment by simonszu** [comment]
I have implemented your annotations. I am not sure about squashing this commit into the last one. On the one hand the changes were very small and do not need to appear in the global commit log, on the other hand the commit message grew bigger. I didn't want to check if i can squash a commit into another commit which was not directly before the to-squash commit, so it is how it is.

**Comment by igneus** [comment]
No squashing necessary, I will be happy to merge as is - but there is currently one failing spec (see the link to Travis CI below).

**Comment by simonszu** [comment]
I see. It is strange, since `rake spec` runs without errors on my machine. In the output, travis says it fails on the check when querying an invalid file from the file system, it should output "Invalid file format". When manually executing this query command, i get "Invalid file format" as well. 

So i really cannot understand why Travis fails, i'm sorry. Maybe you can pull this pull request on your machine and run the spec manually for yourself, so we can determine if it is Travis' fault or my machine's fault.

**Comment by igneus** [comment]
See the report of the failed expectation once more: the expected output is there, but it is preceded by a deprecation warning. The warning is being printed to the STDERR, so I suggest to check only standard output (exposed by aruba through method `all_stdout`) instead of all output.

(We should really get rid of the warning, too, but that's out of the scope of this PR.)

**Comment by simonszu** [comment]
Well, the die-Message is printed to STDERR as well (see https://github.com/igneus/calendarium-romanum/blob/master/lib/calendarium-romanum/cli.rb#L107). Should i change it there as well?

If not, i fear that the tests check for standard output, and fail, since the error messages are sent to STDERR, and STDOUT stays empty.

**Comment by igneus** [comment]
Stupid me, I didn't realize we print error output to STDERR. (And we definitely don't want to change anything about this.) Then I would change the expectation from `... start_with ...` to `... include ...` or `... end_with ...`.

**Comment by simonszu** [comment]
No problem. Here you go. ;)

**Comment by igneus** [comment]
Thanks, merging.

**Pull request merged**